test/test_model.py

```diff
@@ -175,6 +181,49 @@ def test_config(self):
             # Confirm that we cannot modify config values.
             self.model.config['foo'] = 'bar'
 
+    def test_is_leader(self):
+        self.backend = op.model.ModelBackend()
+        self.model = op.model.Model('myapp/0', ['db0', 'db1', 'db2'], self.backend)
+
+        def check_remote_units():
+            fake_script(self, 'relation-ids',
+                        """[ "$1" = db1 ] && echo '["db1:4"]' || echo '[]'""")
+
+            fake_script(self, 'relation-list',
+                        """[ "$2" = 4 ] && echo '["remoteapp1/0", "remoteapp1/1"]' || exit 2""")
+
+            # Cannot determine leadership for remote units.
+            for u in self.model.get_relation('db1').units:
+                with self.assertRaises(RuntimeError):
+                    u.is_leader
+
+        fake_script(self, 'is-leader', 'echo true')
+        self.assertTrue(self.model.unit.is_leader)
+
+        check_remote_units()
+
+        self.assertEqual(fake_script_calls(self), [
```

Comment:
This is not necessary here as it's being tested once more at the end.

Reply:
It is unlikely that after the first part of the test `fake_script_calls(self)` will return an empty list and after the second part it will return a duplicated set of calls so I could remove this. However, doing only one check at the end of the test is not equivalent to what I have now.

In #52 I suggested to add a read-and-clear capability to `fake_script_calls` to avoid making that judgement in less trivial cases than this one.

```
def fake_script_calls(test_case, clear=False):
    with open(test_case.fake_script_path / 'calls.txt', 'r+') as f:
        calls = [line.split(';') for line in f.read().splitlines()]
        if clear:
            f.truncate(0)

        return calls
```

I could just include it in this PR and use it here.